### PR TITLE
Fixed some issues regarding generation for Jersey2 library with Java8

### PR DESCRIPTION
**Pull request by mamghari**
Hi Everyone,
This is my first pull request so please be nice with me :)
If I can improve something feel free to comment it there.
Thank you,
Mohamed

### PR checklist

- [x] Read the [contribution guildelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [ ] Ran the shell/batch script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates)
- [x] Filed the PR against the correct branch: master for non-breaking changes and `2.3.0` branch for breaking (non-backward compatible) changes.

### Description of the PR

(details of the change, additional tests that have been done, reference to the issue for tracking, etc)

### CODE REVIEW

**Comment by wing328** [comment]
Thanks for the PR but your commit (as shown in the Commits tab) is not linked to your Github account, which means this PR won't count as your contribution in https://github.com/swagger-api/swagger-codegen/graphs/contributors.

Let me know if you need help fixing it.

Ref: https://github.com/swagger-api/swagger-codegen/wiki/FAQ#how-can-i-update-commits-that-are-not-linked-to-my-github-account

**Comment by wing328** [comment]
The CI reports the following issue:
```
$ /bin/bash ./bin/utils/detect_carriage_return.sh
modules/swagger-codegen/src/main/resources/JavaJaxRS/JodaLocalDateProvider.mustache
modules/swagger-codegen/src/main/resources/JavaJaxRS/apiServiceFactory.mustache
modules/swagger-codegen/src/main/resources/JavaJaxRS/JodaDateTimeProvider.mustache
modules/swagger-codegen/src/main/resources/JavaJaxRS/jacksonJsonProvider.mustache
Templates contain carriage return '/r'. Please remove it and try again.
```
which means the files listed above contains Windows carriage return. Please remove it and try again.

File: modules/swagger-codegen/src/main/resources/JavaJaxRS/apiServiceFactory.mustache [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
+package {{package}}.factories;
+
+import {{package}}.{{classname}}Service;
+import {{package}}.impl.{{classname}}ServiceImpl;
+
+{{>generatedAnnotation}}
+public class {{classname}}ServiceFactory {
+    private final static {{classname}}Service service = new {{classname}}ServiceImpl();
+
+	private {{classname}}ServiceFactory(){
```
Comment (wing328) [inline modules/swagger-codegen/src/main/resources/JavaJaxRS/apiServiceFactory.mustache]: @mamghari Please replace tab with 4-space instead.

Comment (mamghari) [inline modules/swagger-codegen/src/main/resources/JavaJaxRS/apiServiceFactory.mustache]: Hi @wing328 
thank  you for your review. I am doing the changes.

Thanks

**Comment by wing328** [comment]
Seems like the rebase has failed. I would suggest to create a new branch and `cherry-pick` the commits related to this PR, and then submit a new PR instead.

**Comment by mamghari** [comment]
As suggested I will create another branch and resubmit the PR.
Thank you.

**Pull request closed without merging**